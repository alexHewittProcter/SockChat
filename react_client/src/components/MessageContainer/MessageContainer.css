.message-input-bar > .input-group > *,
.message-input-bar > .input-group > * > .btn {
  border-radius: 0;
}

.message-list-wrapper {
  height: calc(100% - 38px);
  position: relative;
}

#message-list {
  margin: 0;
  padding: 0;
  max-height: 100%;
  padding-bottom: 5px;
  overflow: auto;
}

.message {
  text-align: left;
  padding: 5px 20px;
}

.message:nth-child(even) {
  background-color: lightgray;
}

.message-container {
  position: absolute;
  height: 100%;
  width: 100%;
}
