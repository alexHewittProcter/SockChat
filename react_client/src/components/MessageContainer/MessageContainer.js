import React from 'react';
import io from 'socket.io-client';
import MessageListComponent from '../MessageList/MessageList';
import MessageInputBar from '../MessageInputBar/MessageInputBar';
import './MessageContainer.css';

export class MessageContainerComponent extends React.Component {
  constructor(props) {
    super(props);
    this.state = { messages: [], client: null };
    this.sendMessage = this.sendMessage.bind(this);
  }
  render() {
    return (
      <div className="message-container">
        <MessageListComponent messages={this.state.messages} />
        <MessageInputBar messageSend={this.sendMessage} />
      </div>
    );
  }
  componentDidMount() {
    if (!this.state.client) {
      const sockClient = io('http://192.168.0.36:4000');
      sockClient.on('connect', () => console.log('connected'));
      sockClient.on('chat message', (msg) => {
        this.setState((prevState) => ({
          messages: [...prevState.messages, msg],
        }));
      });
      this.setState({ client: sockClient });
    }
  }
  componentWillUnmount() {
    this.setState((prevState) => {
      if (prevState.client) prevState.client.disconnect();
      return { client: null };
    });
  }
  sendMessage(message) {
    if (this.state.client) this.state.client.emit('chat message', message);
  }
}

export default MessageContainerComponent;
