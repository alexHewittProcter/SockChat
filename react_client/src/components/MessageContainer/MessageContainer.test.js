import React from 'react';
import { shallow } from 'enzyme';
import { MessageContainerComponent } from './MessageContainer';

describe('MessageContainer', () => {
  let messageContainer;
  let mockSendMessage;
  let storeProps;

  beforeEach(() => {
    mockSendMessage = jest.fn();
    storeProps = { messages: [], sendMessage: mockSendMessage };
    messageContainer = shallow(<MessageContainerComponent {...storeProps} />);
  });

  afterEach(() => {
    messageContainer.unmount();
  });

  it('should render component', () => {
    expect(messageContainer).toMatchSnapshot();
  });
  describe('class methods', () => {
    it('should emit the passed message when sendMessage is called', () => {
      const message = `Hello there`;
      messageContainer.instance().sendMessage(message);
      expect(mockSendMessage).toHaveBeenCalledWith(message);
    });
  });
});
