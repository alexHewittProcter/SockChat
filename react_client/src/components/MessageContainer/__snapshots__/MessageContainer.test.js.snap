// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageContainer should render component 1`] = `
<div
  className="message-container"
>
  <MessageListComponent
    messages={Array []}
  />
  <MessageInputBar
    messageSend={[Function]}
  />
</div>
`;
